### PR TITLE
Table: fix indexing and copy

### PR DESCRIPTION
**Pull request by s-alexey**
Refs #1052. 
New indices for every new table.
Copy domain in Table.copy.

### CODE REVIEW

File: Orange/data/table.py [hunk 1]
```diff
@@ -963,6 +959,8 @@ def copy(self):
         Return a copy of the table
         """
         t = self.__class__(self)
+        t.domain = Domain(t.domain.attributes, t.domain.class_vars,
+                          t.domain.metas, getattr(t.domain, 'source', None))
```
Comment (s-alexey) [inline Orange/data/table.py]: Or should I use `copy.copy(t.domain)`?

File: Orange/data/table.py [hunk 2]
```diff
@@ -11,7 +11,6 @@
 from tempfile import NamedTemporaryFile
 from urllib.parse import urlparse, unquote as urlunquote
 from urllib.request import urlopen
-from urllib.error import URLError
```
Comment (kernc) [inline Orange/data/table.py]: I think this might have been meant to be caught in the `from_url` method. :worried:

Comment (s-alexey) [inline Orange/data/table.py]: But it's unused (caught outside `from_url` in `OWFile`). 
Actually with `from .instance import *` it's a mess here. This import shadows `chain`, `Real` and `Integral` and imports `numpy`.

Comment (kernc) [inline Orange/data/table.py]: > But it's unused

It just might have been meant for that. Indeed, doesn't really hurt if we just let urllib complain more verbosely than we would probably know to.

> it's a mess here.

**Comment by astaric** [comment]
I am not really sure what you are trying to accomplish here.

We tend to consider domain as immutable, any modifications are usually done by constructing a new domain. (i.e. rearranging columns, moving variables from metas to attributes, ...). This is why it was not copied in Table.copy.

Ids work in a similar way. They allow us to use rows from a discretised, filtered table as selections on original data (i.e. using ScatterPlot). Generating new ids for each selection would break this feature.

**Pull request closed without merging**